interfaces/matlab_experimental/Reactor/ReactorBase.m

```diff
@@ -0,0 +1,237 @@
+
+        function r = ReactorBase(id)
+            % ReactorBase Class ::
+            %
+            % :param id:
+            %    ID of reactor object instantiated by specialization.
+
+            arguments
+                id (1,1) {mustBeInteger, mustBeGreaterThan(id, -1)}
+            end
+
+            r.id = id;
+            phaseID = ctFunc('reactor_phase', id);
+            r.phase= Solution(phaseID);
+        end
+
+        %% ReactorBase Class Destructor
+
+        function delete(r)
+            % Delete the :mat:class:`ReactorBase` object.
+            ctFunc('reactor_del', r.id);
+        end
+
+        %% ReactorBase Utility Methods
+
+        function addSensitivityReaction(r, m)
+            % Specifies that the sensitivity of the state variables with
+            % respect to reaction m should be computed. The reactor must be
+            % part of a network first. ::
+            %
+            %     >> r.addSensitivityReaction(m)
+            %
+            % :param m:
+            %    Index number of reaction.
+
+            ctFunc('reactor_addSensitivityReaction', r.id, m);
+        end
+
+        %% ReactorBase Get Methods
+
+        function typ = get.type(r)
+            typ = ctString('reactor_type', r.id);
+        end
+
+        function name = get.name(r)
+            name = ctString('reactor_name', r.id);
+        end
+
+        function temperature = get.T(r)
+            temperature = ctFunc('reactor_temperature', r.id);
+        end
+
+        function pressure = get.P(r)
+            pressure = ctFunc('reactor_pressure', r.id);
+        end
+
+        function rho = get.D(r)
+            rho = ctFunc('reactor_density', r.id);
+        end
+
+        function mass = get.M(r)
+            mass = ctFunc('reactor_mass', r.id);
+        end
+
+        function volume = get.V(r)
+            volume = ctFunc('reactor_volume', r.id);
+        end
+
+        function enthalpy_mass = get.H(r)
+            enthalpy_mass = ctFunc('reactor_enthalpy_mass', r.id);
+        end
+
+        function intEnergy_mass = get.U(r)
+            intEnergy_mass = ctFunc('reactor_intEnergy_mass', r.id);
+        end
+
+        function a = get.area(r)
+            a = ctFunc('reactor_area', r.id);
+        end
+
+        function yi = massFraction(r, species)
+            % Get the mass fraction of a species.
+            %
+            % :param species:
+            %    String/Char array name of the species.
+
+            k = r.phase.speciesIndex(species) - 1;
+            yi = ctFunc('reactor_massFraction', r.id, k);
+        end
+
+        function massFractions = get.Y(r)
+            massFractions = ctArray('reactor_massFractions', r.phase.nSpecies, r.id);
+        end
+
+        function flag = get.chemistryEnabled(r)
+            flag = ctFunc('reactor_chemistryEnabled', r.id);
+        end
+
+        function flag = get.energyEnabled(r)
+            flag = ctFunc('reactor_energyEnabled', r.id);
+        end
```

Comment:
These methods aren't actually defined on the C++ `ReactorBase` class, so I have to assume that calling these raises an exception.